### ISSUE TITLE
Sourcery Updates, SwiftyMocky Breaks. 🎵 

### ISSUE THREAD

**Issue by fssilva** [MakeAWishFoundation/SwiftyMocky#281]
Hi, I've been experiencing some broken mocks every time Sourcery releases an update. As of today SwiftyMocky doesn't work with Sourcery `1.3.1`, it generates an invalid `main.swift`

Could we be more specific to which version of Sourcery we depend on ? Having it as `s.dependency 'Sourcery', '>= 1.0.2'` doesn't seems very reliable.

**Comment by g-amichnia** [comment]
@fssilva Hi, could you specify the problem? I will try it with new sorcery, but meantime, can you check few things?

1. Do you use CLI for generation? If not, I highly recommend, since it makes your life way way easier.
2. If not, do you have correct template directory? It should be:

```yml
templates:
  - ./Pods/SwiftyMocky/Sources/SwiftyMocky
```

From problem description it looks like you are using:

```yml
templates:
  - ./Pods/SwiftyMocky/Templates
```

which contains partial templates (we use that folder for development). I think we will need to rethink location and naming of that.

**Comment by g-amichnia** [comment]
OK, seems that anyway some api in Sourcery changed. I was expecting no breaking changes when only minor changes, but seems it is untrue. I will look when this change was introduced, and release SM fix.

**Comment by fssilva** [comment]
Hi, 

1. No, I use the Xcode Build Script for generation. 
2. I'm using the correct template. `- ./Pods/SwiftyMocky/Sources/SwiftyMocky`

The point I wanna make here is that Sourcery is a very important piece in SwifyMocky. Changes on Sourcery have a huge impact on the project and therefore should be carefully tested before making it automatically available to everyone.

Instead of having this`Sourcery', '>= 1.0.2'` it should be `Sourcery', '1.0.2'` (or whatever version has been tested). Releasing a new fix without pinning the version will not solve the problem, but only delay it.

**Comment by g-amichnia** [comment]
@fssilva yes, you are completely right. I think I was over optimistic assuming Sourcery will keep same api's and behaviour after 1.0 release. They did a lot of cool stuff recently, I need to prepare SM for Sourcery 1.3.

I will be more cautious from now on. Anyway, this issue should be fixed with our recent 4.0.4 release.
